src/main/java/io/confluent/connect/hdfs/wal/FSWAL.java

```diff
@@ -102,6 +103,20 @@ public void apply() throws ConnectException {
       if (!storage.exists(logFile)) {
         return;
       }
+
+      // An empty WAL might be created (connect process crashed before the WAL header is written).
+      // This can cause the recovery to persistently fail.
+      FileStatus[] statuses = storage.listStatus(logFile);
+
+      if (statuses.length != 1) {
+        throw new ConnectException("Expected only one WAL file: " + logFile);
+      }
+
+      if (statuses[0].getLen() == 0) {
+        log.warn("Skip empty WAL file: " + logFile);
+        return;
```

Comment:
Wouldn't we still want to acquire the lease here? Shouldn't we be acquiring if for no other reason than to make sure that we *can* at this point? Otherwise, even if its zero bytes now, couldn't something else continue write entries into the WAL.